### PR TITLE
Allow adding a custom validation error at a given array index

### PR DESCRIPTION
**Pull request by knilink**
For #223

### CODE REVIEW

**Comment by knilink** [comment]
Hum, it looks like the test case needs to be updated as well since the errorSchema is changed.
But I'm not sure whether it's appropriate for me to update the script to test my own code.

File: src/validate.js [hunk 1]
```diff
@@ -95,6 +95,11 @@ function createErrorHandler(formData) {
       return {...acc, [key]: createErrorHandler(formData[key])};
     }, handler);
   }
+  if(Array.isArray(formData)){
+    return Object.assign([],formData.reduce((acc, item, index) => {
```
Comment (n1k0) [inline src/validate.js]: I'm not sure `Object.assign` works with arrays; the `reduce` result should suffice in theory.

**Comment by n1k0** [comment]
Thanks for the attempt at tackling this! Tests should be updated in `validate_test.js`, where new ones should be added for this functionality too :)

File: src/validate.js (hunk 1 above)
Comment (n1k0) [inline src/validate.js]: Ohh, I see what you did here, smart; we need keys as object properties. That should actually work.

**Comment by leplatrem** [comment]
Thanks for your intent! I'm closing this PR for lack of response. Feel free to reopen with tests and more details about relevance of use case.

> [About Closing Pull Requests](https://github.com/blog/2124-kindly-closing-pull-requests)

**Pull request closed without merging**